### PR TITLE
Improve documentation of device authentication

### PR DESCRIPTION
**Pull request by sophokles73**
The (detailed) description of how devices are authenticated using the
information on record has been moved to the Device Identity concept
page's Authentication section in order to reduce text duplication.

The protocol adapter user guides now only describe the binding to the
concrete transport protocol.

### CODE REVIEW

File: site/documentation/content/user-guide/amqp-adapter.md [hunk 1]
```diff
@@ -9,54 +9,51 @@ messages to Eclipse Hono&trade;'s Telemetry, Event and Command & Control endpoin
...
+[RFC 4616, Section 2](https://datatracker.ietf.org/doc/html/rfc4616#section-2) when connecting to the AMQP adapter.
+The username provided in the *authcid* field must match the pattern *auth-id@tenant*, e.g. `sensor1@DEFAULT_TENANT`.
 
-The adapter verifies the credentials provided by the client against the credentials that the
-[configured Credentials service]({{< relref "/admin-guide/common-config#credentials-service-connection-configuration" >}}) has on record for the client.
-If the credentials match, the client device can proceed to publish messages to Hono.
+The adapter then verifies the provided username and password against the credentials that the
+[configured Credentials service]({{< relref "/admin-guide/common-config#credentials-service-connection-configuration" >}})
+has on record for the client as described in
+[Username/Password based Authentication]({{< relref "/concepts/device-identity#username-password-based-authentication" >}}).
```
Comment (calohmn) [inline site/documentation/content/user-guide/amqp-adapter.md:34]: Link anchor doesn't match - `relref` should be be `/concepts/device-identity#usernamepassword-based-authentication`.
This needs to be fixed in the corresponding chapters in the other protocol adapter guides as well.

File: site/documentation/content/user-guide/amqp-adapter.md [hunk 2]
```diff
@@ -451,40 +514,46 @@ Command response sent [outcome: Accepted{}]
...
-Note that of the standard AMQP 1.0 message properties only the *content-type* and *ttl* can be set this way to a default value.
+Note that of the standard AMQP 1.0 message properties only the *content-type* and *ttl* can be set this way to a
+default value.
 
 ### Event Message Time-to-live
 
-Events published by devices will usually be persisted by the AMQP Messaging Network in order to support deferred delivery to downstream consumers.
-In most cases the AMQP Messaging Network can be configured with a maximum *time-to-live* to apply to the events so that the events will be removed
-from the persistent store if no consumer has attached to receive the event before the message expires.
+Events published by devices will usually be persisted by the messing infrastructure in order to support deferred
```
Comment (calohmn) [inline site/documentation/content/user-guide/amqp-adapter.md:541]: `messaging infrastructure`
This needs to be fixed in the corresponding chapters in the other protocol adapter guides as well.

**Comment by sophokles73** [comment]
@calohmn I have pushed fixes

**Review by calohmn — APPROVED** [review]
LGTM

**Pull request merged**